drivers/led/led_gpio.c

```diff
@@ -92,6 +101,18 @@ static int gpio_led_set_no_delay(struct led_gpio_data *data, bool state)
 
 static int gpio_led_set(struct led_gpio_data *data, bool state)
 {
+#if defined(CONFIG_LED_SOFTWARE_BLINK)
+	if (atomic_test_bit(&data->flags, LED_SW_BLINK)) {
+		data->next_state = state;
+
+		atomic_set_bit(&data->flags, LED_BLINK_DISABLE);
+		k_timer_start(&data->timer, K_NO_WAIT, K_NO_WAIT);
```

Comment:
This is here to avoid some race conditions when during the software blink is used, the led migth be set to different state from a higher priority interrupt or thread than a timer handler also the set function can be called when timer is pending. So setting the LEDs from one context and using the atomic flags eliminates this problem.